platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkJavaCompiler.java

```diff
@@ -62,15 +67,20 @@ public WorkResult execute(JavaCompileSpec spec) {
         LOGGER.info("Compiling with JDK Java compiler API.");
 
         ApiCompilerResult result = new ApiCompilerResult();
-        JavaCompiler.CompilationTask task = createCompileTask(spec, result);
+        JavaCompiler.CompilationTask task;
+        try {
+            task = createCompileTask(spec, result);
+        } catch (RuntimeException ex) {
```

Comment:
🤔  I was thinking maybe we should only catch a more specific exceptions, but reading the code now makes me think this is the correct. We get the a `result` object when the compiler was invoked and did some work. When the compiler cannot be invoked, we always get an exception which we capture and rethrow.

Reply:
This is the core which we can build upon. We can start out with this pretty rudimentary solution for now, and build up more and more details as we understand what can come out of the compiler.